onnxscript/rewriter/broadcast_to_matmul.py

```diff
@@ -31,7 +31,9 @@ def check_if_need_reshape(match_bindings: dict[str, ir.Value | Any]) -> bool:
     """
     input_a_shape = match_bindings["input_a"].shape
     input_b_shape = match_bindings["input_b"].shape
-    shape_c = match_bindings["shape_c"].value_as_np_array
+    # TODO: Get a helper func to get const_value
+    shape_c_value = _ir_utils.propagate_const_value(match_bindings["shape_c"])
```

Comment:
Trying to understand this ... I think I would push this into the rewriter engine, instead of burdening the rule-writer. Ideally, we would have a separate pass to compute constant-values (eg., as done by the optimizer). Then, whenever the rewriter modifies the graph, it can invoke computation of constant values on newly inserted nodes? For now, we can probably do the second step? (first is a separate issue, anyway, and requires migration of optimizer to the new IR).

Reply:
The utils under _ir_utils are all temporary hack, and wait for the implementation in IR. I think this will be removed once we have propagation in IR.

Reply:
https://github.com/microsoft/onnxscript/issues/1404